presto-product-tests/src/main/java/io/prestosql/tests/hive/TestRowByRowDelete.java

```diff
@@ -0,0 +1,264 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.prestosql.tests.hive;
+
+import com.google.common.util.concurrent.Uninterruptibles;
+import io.airlift.log.Logger;
+import io.prestosql.tempto.ProductTest;
+import org.testng.annotations.Test;
+
+import java.time.Duration;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+
+import static com.google.common.base.Preconditions.checkArgument;
+import static io.prestosql.tempto.assertions.QueryAssert.Row.row;
+import static io.prestosql.tempto.assertions.QueryAssert.assertThat;
+import static io.prestosql.tests.TestGroups.HIVE_TRANSACTIONAL;
+import static io.prestosql.tests.utils.QueryExecutors.onHive;
+import static io.prestosql.tests.utils.QueryExecutors.onPresto;
+import static java.lang.String.format;
+
+public class TestRowByRowDelete
```

Comment:
Call this class `TestDeleteHiveTransactionalTable` for easier finding.
(then `TestHiveTransactionalTable` should be renamed to `TestReadHiveTransactionalTable` perhaps)

btw "row by row delete" is often called "row-level delete"

Reply:
Row-level delete is indeed than row-by-row delete.

Right now, there is a test in the class for metadata delete for non-transactional tables.  Maybe the name should be TestHiveRowLevelDelete.

*********************

Thanks very much for the pointers, @findepi!

Reply:
I settled on TestHiveTransactionalDelete.